openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -256,7 +256,15 @@ def is_flag_active(self, flag_name, check_before_waffle_callback=None, flag_unde
                         value = flag_undefined_default
 
                 if value is None:
-                    value = flag_is_active(get_request(), namespaced_flag_name)
+                    request = crum.get_current_request()
+                    if request:
+                        value = flag_is_active(request, namespaced_flag_name)
+                    else:
+                        log.warn(u"%sFlag '%s' accessed without a request", self.log_prefix, namespaced_flag_name)
```

Comment:
1. Did you resolve your issue, or are you trying to use this to resolve it?
2. It seems like we should log an error and not a warning, because I don't see when this would be a normal case.  Unless you found that this is a normal case.  If so, please explain.
3. Can we create a request with an anonymous user instead?  That way if someone turns the flag on for Everyone in waffle, it will be on in this case as well.

Reply:
1. I'm trying to use this to resolve it. What appears to happen is that when a page redirects to a 404 that CRUM doesn't have a "current request" recorded. So then when we render the 404 page, the Waffle utils logic tries to call Waffle with None as the request. This blows up inside Waffle which doesn't expect this to happen.

2. I didn't want to log an error because that could end up paging DevOps. @feanil do you have a recommendation here? 

3. Hmmm... we could do that. It seems a bit risky because other code might get confused. I considered whether I could do something else to ensure that a request was available but I couldn't think of one. I suppose I could create a temporary one and then delete it again after calling Waffle. Do you feel that this is necessary?

I want to get this merged quickly so that we can start enabling the new responsive features on prod early next week. However I've missed my window for today, so I'm willing to work on it further today and Monday morning. Thanks!

Reply:
Thanks for the explanation @andy-armstrong.  Given your explanation, I would just add something like the following to the comment: "This case seems to occur when a page redirects to a 404.  In this case, we'll just return the default value."  

It may be slightly better to return the default waffle value in case a flag was turned on for everyone, but given the error case, it is probably not a big deal.  Most waffle flags probably wouldn't affect this case, but yours might.  If you had turned it on for everyone in waffle, would you want it on in this case?

Reply:
Thanks, @robrap. I've made the suggested comment change.